config/src/test/java/org/axonframework/config/utils/AssertUtils.java

```diff
@@ -30,20 +32,19 @@ private AssertUtils() {
     /**
      * Assert that the given {@code assertion} succeeds with the given {@code time} and {@code unit}.
      *
-     * @param time      an {@code int} which paired with the {@code unit} specifies the time in which the assertion must
+     * @param duration  specifies the time in which the assertion must
      *                  pass
-     * @param unit      a {@link TimeUnit} in which {@code time} is expressed
      * @param assertion a {@link Runnable} containing the assertion to succeed within the deadline
      */
     @SuppressWarnings("Duplicates")
-    public static void assertWithin(int time, TimeUnit unit, Runnable assertion) {
+    public static void assertRetryingWithin(Duration duration, Runnable assertion) {
```

Comment:
Because `assertion` is run multiply times if it is failing while the execution time is less then the `duration`.
This behaviour can be demonstrated explicitly: Add System.out.println(result of config.getComponent(TokenStore.class)...) in the lambda before `assertEquals` in the `DefaultConfigurerTest.defaultConfigurationWithTrackingProcessorExplicitlyConfigured`